tests/test_backend_github.py

```diff
@@ -61,18 +63,20 @@ def create_project(self):
         self.session.add(project)
         self.session.commit()
 
+        # project homepage with trailing slash
         project = model.Project(
             name='foobar',
-            homepage='http://github.com/foo/bar',
-            version_url='foobar/bar',
+            homepage='http://github.com/foo/bar/',
             backend=BACKEND,
         )
         self.session.add(project)
         self.session.commit()
 
+        # project homepage on GitHub but not project root
+        # should correctly match only needed segment
         project = model.Project(
             name='pkgdb2',
-            homepage='https://github.com/fedora-infra/pkgdb2',
+            homepage='https://github.com/fedora-infra/pkgdb2/issues',
```

Comment:
I am wondering if we want to support this, as it is not a valid home page and if we start to be smart and cover corner cases such as this one, we may end up with a large bowl of spaghetti hard to maintain in the long term.